src/blueteam/Gardener.java

```diff
@@ -29,33 +38,50 @@
 	boolean findSpot(float radius) {
 		// get robots in radius
 		RobotInfo[] nearbyRobots = rc.senseNearbyRobots(radius);
-		for (RobotInfo robot : nearbyRobots) {
-			// check for archrons and gardeners around
-			//TODO make cleverer location finding
-			if (robot.getType().equals(RobotType.GARDENER) || robot.getType().equals(RobotType.ARCHON)) {
-				Direction dir;
-				// TODO use constants
-				// go away from gardener/archon or go in direction of enemy
-				// TODO is Math.random best way ?
-				if (Math.random() > 0.2) {
-					if (Math.random() > 0.5)
-						dir = rc.getLocation().directionTo(rc.getInitialArchonLocations(enemy)[0])
-								.rotateLeftDegrees((float) Math.random() * 60);
-					else
-						dir = rc.getLocation().directionTo(rc.getInitialArchonLocations(enemy)[0])
-								.rotateRightDegrees((float) Math.random() * 60);
-				} else
-					dir = rc.getLocation().directionTo(robot.getLocation()).opposite();
-				// get away from other gardeners or archons in somehow random direction
-				// TODO use constants
-				dir = dir.rotateLeftDegrees((float) Math.random() * 60 - 30);
-				// black dot in debugging
-				rc.setIndicatorDot(rc.getLocation().add(dir), 1, 1, 1);
-				tryMove(dir);
-				return false;
+
+		if (roundCounter > 100) {
+			roundCounter = 0;
+			currentDir = randomDirection();
+		}
+		boolean rdyToBuild = true;
+		//check for robots
+		for (RobotInfo robot : nearbyRobots)
+			if (robot.getTeam().equals(rc.getTeam()))
+				if (robot.getType().equals(RobotType.GARDENER) || robot.getType().equals(RobotType.ARCHON))
+					rdyToBuild = false;
+		if (!rdyToBuild) {
+			try {
+				if (!rc.onTheMap(rc.getLocation(), radius)) {
+					rdyToBuild = false;
+					roundCounter += 5;
+				}
+
+				boolean success = false;
+				int angle = 10;
+				Direction newWay = currentDir;
+
+				// First, try intended direction
+				if (rc.canMove(currentDir)) {
+					rc.move(currentDir);
+					success = true;
+				}
+				// Now try a bunch of similar angles
+				while (!success) {
+					newWay = currentDir.rotateLeftDegrees(rand.nextInt(angle) - angle / 2);
+					// Try the offset of the left side
+					if (rc.canMove(newWay)) {
+						rc.move(newWay);
+						success = true;
+					}
+					angle += 4;
+				}
+				currentDir = newWay;
+				// A move never happened, so return false.
```

Comment:
It should be more polished and divided into methods, will look into it asap. Your suggestion can be the right way.